src/main/java/gui/EditRulePopup.java

```diff
@@ -0,0 +1,277 @@
+		createAddMetricPanel();
+
+		metricsPanel = new JPanel();
+		metricsPanel.setLayout(new BorderLayout());
+		metricsPanel.add(addNewMetricPanel, BorderLayout.NORTH);
+		metricsPanel.add(metricsScrollpane, BorderLayout.CENTER);
+		return metricsPanel;
+	}
+
+	/**
+	 * Creates and returns the JPanel responsible for holding the JButtons which
+	 * allow all metrics to be cleared, or the rule to be saved.
+	 * 
+	 * @return
+	 */
+	private JPanel createControlPanel() {
+		controlPanel = new JPanel();
+		controlPanel.setLayout(new GridLayout(1, 3));
+		
+		
+		JButton clearButton = new JButton("Clear Metrics");
+		JButton deleteButton = new JButton("Delete Rule");
+		JButton saveButton = new JButton("Save Rule");
+		
+		
+		clearButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				System.out.println("Clear button pressed!");
+				clearMetricsListPanel();
+			}
+		});
+		
+		deleteButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (nameText.getText().isEmpty()) {
+					JOptionPane.showMessageDialog(null, "Please insert a rule name!");
+				} else {
+					System.out.println("Delete button pressed!");
+					System.out.println("Rule name: " + nameText.getText());
+				}
+			}
+		});
+		
+		saveButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (nameText.getText().isEmpty()) {
+					JOptionPane.showMessageDialog(null, "Please insert a rule name!");
+				} else {
+					System.out.println("Save button pressed!");
+					System.out.println("Rule name: " + nameText.getText());
+				}
+			}
+		});
+		
+		controlPanel.add(clearButton);
+		controlPanel.add(deleteButton);
+		controlPanel.add(saveButton);
+		return controlPanel;
+
+	}
+	
+	/**
+	 * This method both returns and creates the JPanel holding the line which allows
+	 * users to add new metrics into the metrics list. TODO: The currently
+	 * hard-coded values of the JBoxes are to be replaced with the correct, final
+	 * values once the ENUMs are created.
+	 * 
+	 * @return
+	 */
+	public JPanel createAddMetricPanel() {
+		addNewMetricPanel.removeAll();
+		addNewMetricPanel.setLayout(new GridLayout(1, 6));
+		condition = new JComboBox<>();
+		setConditionVisibility();
+		JLabel ifCondition = new JLabel("IF", SwingConstants.CENTER);
+		JComboBox<String> value = new JComboBox<>();
+		value.addItem("LOC");
+		value.addItem("MET");
+		
+		JComboBox<String> comparison = new JComboBox<>();
+		comparison.addItem(">");
+		comparison.addItem("<");
+		comparison.addItem("=");
+		comparison.addItem("!=");
+		
+		JTextField threshold = new JTextField("10");
+		
+		JButton addMetricButton = new JButton("Add");
+		addMetricButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (value.getItemCount() == 0) {
+					JOptionPane.showMessageDialog(null, "Please check if all metric fields were entered correctly!");

```

Comment:
seems like a weird thing to say, metrics aren't entered by the user, they're predetermined by us. I think this may be unnecessary?

Reply:
The metrics themselves are predetermined by us, but the threshold is user input, which is what this _if_ **should**  be guarding against. The _if_ as it currently stands is wrong, however. It served to check if the selection box had something selected in earlier stages in the development process though, and is quite useless now, as it's impossible to have it empty. I'll remove it when I get the chance, and implement a proper validation for the user input.

Reply:
please update this when possible